src/DotLiquid/Util/ObjectExtensionMethods.cs

```diff
@@ -109,6 +109,20 @@ public static bool SafeTypeInsensitiveEqual(this object value, object otherValue
                 return false;
             }
 
+            // NOTE(Rodney Richardson): If either value is boolean or the string "true" or "false", compare with Falsy/Truthy
+            // We can't use IsTruthy() directly, as everything that's not Falsy is considered Truthy

```

Comment:
but that contradicts the Shopify documentation referred to by the IsTruthy method.